backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java

```diff
@@ -172,6 +165,58 @@ public List<FileResponse> getAllFilesWithParent(UUID parentFolderId) {
     return asFileResponseList(allFiles);
   }
 
+  @Override
+  @Transactional(rollbackFor = Exception.class)
+  public FileResponse updateFileContent(MultipartFile multipartFile, UUID fileId) {
+
+    FileMetadataEntity fileMetadata =
+        fileMetadataRepository.findById(fileId).orElseThrow(FileNotFoundException::new);
+
+    checkFileTypeNotChanged(fileMetadata, multipartFile.getContentType());
+
+    String oldFileContentId = fileMetadata.getContentId();
+    fileMetadata.setType(multipartFile.getContentType());
+    fileMetadata.setSize(multipartFile.getSize());
+
+    checkOnlyCreatorsCanAccessPrivateBuckets(fileMetadata);
+    checkOnlyEmployeesCanAccessCompanyBuckets(fileMetadata);
+    checkBucketNotDeleted(fileMetadata);
+    checkBucketSizeExceedsMax(fileMetadata);
+
+    FileMetadataEntity savedFileMetadata = uploadFileContent(fileMetadata, multipartFile);
+    deleteFileContent(oldFileContentId);
+    actionService.recordAction(Collections.singleton(savedFileMetadata), ActionType.EDIT);
+
+    return modelMapper.map(savedFileMetadata, FileResponse.class);
+  }
+
+  private void checkFileTypeNotChanged(FileMetadataEntity fileMetadata, String newFileType) {
+    String oldFileType = fileMetadata.getType();
+    if (!oldFileType.equals(newFileType)) throw new FileCannotChangeTypeException();
+  }
+
+  private FileMetadataEntity uploadFileContent(
+      FileMetadataEntity emptyFileMetadata, MultipartFile multipartFile) {
+    try {
+      InputStream fileContents = multipartFile.getInputStream();
+      FileMetadataEntity savedFileMetadata = storeFileContents(emptyFileMetadata, fileContents);
+      return savedFileMetadata;
+    } catch (IOException e) {
+      log.error(e);
+      throw new FileIOException();
+    }
+  }
+
+  private void deleteFileContent(String objectId) {
+    try {
+      if (objectId == null || objectId.isEmpty()) throw new FileNotFoundException();
```

Comment:
I believe there is a method from StringUtils named isEmpty that does this two checks

Reply:
`isAllBlank()` from `org.apache.commons.lang3.StringUtils`?